gluonnlp/loss/activation_loss.py

```diff
@@ -0,0 +1,129 @@
+      journal={arXiv preprint arXiv:1708.01009},
+      year={2017}
+    }
+
+    Parameters
+    ----------
+    alpha : float, default 0
+        The scaling coefficient of the regularization.
+
+    Inputs:
+        - **states**: the stack outputs from RNN,
+        which consists of output from each time step (TNC).
+
+    Outputs:
+        - **loss**: loss tensor with shape (batch_size,). Dimensions other than
+          batch_axis are averaged out.
+    """
+    def __init__(self, alpha=0, weight=None, batch_axis=None, **kwargs):
+        super(ActivationRegularizationLoss, self).__init__(weight, batch_axis, **kwargs)
+        self._alpha = alpha
+
+    def __repr__(self):
+        s = 'ActivationRegularizationLoss (alpha={alpha})'
+        return s.format(alpha=self._alpha)
+
+    def hybrid_forward(self, F, *states): # pylint: disable=arguments-differ
+        # pylint: disable=unused-argument
+        if not self._alpha:
+            if not states:
+                means = [self._alpha * state.__pow__(2).mean()
+                         for state in states[-1:]]
+                return nd.add_n(*means)
+        return 0
+
+class TemporalActivationRegularizationLoss(Loss):
+    r"""Computes Temporal Activation Regularization Loss. (alias: TAR)
+
+    The formulation is as below:
+
+    .. math::
+
+        L = \beta L_2(h_t-h_{t+1})
+
+    where $L_2(\dot) = {||\dot||}_2$, $h_t$ is the output of the RNN at timestep t,
+     $h_{t+1} is the output of the RNN at timestep t+1, $\beta$ is scaling coefficient.
+
+    The implementation follows the work:
+
+    @article{merity2017revisiting,
+      title={Revisiting Activation Regularization for Language RNNs},
+      author={Merity, Stephen and McCann, Bryan and Socher, Richard},
+      journal={arXiv preprint arXiv:1708.01009},
+      year={2017}
+    }
+
+    Parameters
+    ----------
+    beta : float, default 0
+        The scaling coefficient of the regularization.
+
+    Inputs:
+        - **states**: the stack outputs from RNN,
+        which consists of output from each time step (TNC).
+
+    Outputs:
+        - **loss**: loss tensor with shape (batch_size,). Dimensions other than
+          batch_axis are averaged out.
+    """
+
+    def __init__(self, beta=0, weight=None, batch_axis=None, **kwargs):
+        super(TemporalActivationRegularizationLoss, self).__init__(weight, batch_axis, **kwargs)
+        self._beta = beta
+
+    def __repr__(self):
+        s = 'TemporalActivationRegularizationLoss (beta={beta})'
+        return s.format(beta=self._beta)
+
+    def hybrid_forward(self, F, *states): # pylint: disable=arguments-differ
+        # pylint: disable=unused-argument
+        if not self._beta:
+            if not states:
+                means = [self._beta * (state[1:] - state[:-1]).__pow__(2).mean()
+                         for state in states[-1:]]
+                return nd.add_n(*means)
+        return 0
```

Comment:
native 0 doesn't have backward. what if users just do backward directly on the output of this class? should we use zeros_like(x) instead?

Reply:
sounds good